lib/Net/SAML2/Binding/Redirect.pm

```diff
@@ -264,26 +264,26 @@ sub verify {
 
     my %errors;
     foreach my $crt (@{$self->cert}) {
-            my $cert = Crypt::OpenSSL::X509->new_from_string($crt);
-            my $rsa_pub = Crypt::OpenSSL::RSA->new_public_key($cert->pubkey);
-
-            if ($sigalg eq 'http://www.w3.org/2001/04/xmldsig-more#rsa-sha256') {
-                $rsa_pub->use_sha256_hash;
-            } elsif ($sigalg eq 'http://www.w3.org/2001/04/xmldsig-more#rsa-sha224') {
-                $rsa_pub->use_sha224_hash;
-            } elsif ($sigalg eq 'http://www.w3.org/2001/04/xmldsig-more#rsa-sha384') {
-                $rsa_pub->use_sha384_hash;
-            } elsif ($sigalg eq 'http://www.w3.org/2001/04/xmldsig-more#rsa-sha512') {
-                $rsa_pub->use_sha512_hash;
-            } elsif ($sigalg eq 'http://www.w3.org/2000/09/xmldsig#rsa-sha1') {
-                $rsa_pub->use_sha1_hash;
-            } else {
-                $errors{$cert->fingerprint_sha256} = "Unsupported Signature Algorithim: $sigalg";
-                warn "Unsupported Signature Algorithim: $sigalg";
-            }
-                if (!$rsa_pub->verify($signed, $sig)) {
-                    $errors{$cert->fingerprint_sha256} = "Bad Signature " . encode_base64($sig, '');
-                }
+        my $cert = Crypt::OpenSSL::X509->new_from_string($crt);
+        my $rsa_pub = Crypt::OpenSSL::RSA->new_public_key($cert->pubkey);
+
+        if ($sigalg eq 'http://www.w3.org/2001/04/xmldsig-more#rsa-sha256') {
+            $rsa_pub->use_sha256_hash;
+        } elsif ($sigalg eq 'http://www.w3.org/2001/04/xmldsig-more#rsa-sha224') {
+            $rsa_pub->use_sha224_hash;
+        } elsif ($sigalg eq 'http://www.w3.org/2001/04/xmldsig-more#rsa-sha384') {
+            $rsa_pub->use_sha384_hash;
+        } elsif ($sigalg eq 'http://www.w3.org/2001/04/xmldsig-more#rsa-sha512') {
+            $rsa_pub->use_sha512_hash;
+        } elsif ($sigalg eq 'http://www.w3.org/2000/09/xmldsig#rsa-sha1') {
+            $rsa_pub->use_sha1_hash;
+        } else {
+            $errors{$cert->fingerprint_sha256} = "Unsupported Signature Algorithim: $sigalg";
+            warn "Unsupported Signature Algorithim: $sigalg";
+        }
+        if (!$rsa_pub->verify($signed, $sig)) {
```

Comment:
This is ok with multiple signing keys. It supports key rollover. You advertise two keys. Sign it with the old one. Then flip the key sign it with the new one and your rollover is complete after which the old key is removed. If you die, this defauts the purpose. As soon as we've verified we should return an ok and we are done.